src/main/kotlin/net/leanix/githubagent/config/WebSocketClientConfig.kt

```diff
@@ -0,0 +1,58 @@
+package net.leanix.githubagent.config
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import io.github.resilience4j.retry.annotation.Retry
+import net.leanix.githubagent.handler.BrokerStompSessionHandler
+import net.leanix.githubagent.services.AuthService
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import org.springframework.messaging.converter.MappingJackson2MessageConverter
+import org.springframework.messaging.simp.stomp.StompHeaders
+import org.springframework.messaging.simp.stomp.StompSession
+import org.springframework.scheduling.concurrent.ThreadPoolTaskScheduler
+import org.springframework.web.socket.WebSocketHttpHeaders
+import org.springframework.web.socket.client.standard.StandardWebSocketClient
+import org.springframework.web.socket.messaging.WebSocketStompClient
+import org.springframework.web.socket.sockjs.client.SockJsClient
+import org.springframework.web.socket.sockjs.client.WebSocketTransport
+
+@Configuration
+class WebSocketClientConfig(
+    private val brokerStompSessionHandler: BrokerStompSessionHandler,
+    private val objectMapper: ObjectMapper,
+    private val authService: AuthService,
+    private val leanIXProperties: LeanIXProperties,
+    private val gitHubEnterpriseProperties: GitHubEnterpriseProperties
+) {
+
+    @Retry(name = "ws-init-session")
+    fun initSession(): StompSession {
+        val headers = WebSocketHttpHeaders()
+        val stompHeaders = StompHeaders()
+        stompHeaders["Authorization"] = "Bearer ${authService.getBearerToken()}"
+        stompHeaders["GitHub-Enterprise-URL"] = gitHubEnterpriseProperties.baseUrl
+        stompHeaders["GiHub-Enterprise-App-Id"] = gitHubEnterpriseProperties.gitHubAppId
```

Comment:
I think we need to send the agent version here, not the app id

Reply:
@henriqamaral when you say the agent version, you are referring to the one that is part of the gradle file?

Reply:
Hi @geoandri,
Yes, like we had on the old agents(brokers)